src/psyclone/configuration.py

```diff
@@ -459,14 +473,58 @@ def filename(self):
         '''
         return self._config_file
 
+    @property
+    def kernel_output_dir(self):
+        '''
+        :returns: the directory to which to write transformed kernels.
+        :rtype: str
+        '''
+        if not self._kernel_output_dir:
+            # We use the CWD if no directory has been specified
+            import os
+            self._kernel_output_dir = os.getcwd()
+        return self._kernel_output_dir
+
+    @kernel_output_dir.setter
+    def kernel_output_dir(self, value):
+        '''
+        Setter for kernel output directory.
+        :param str value: directory to which to write transformed kernels.
+        '''
+        self._kernel_output_dir = value
+
+    @property
+    def kernel_naming(self):
+        '''
+        :returns: what naming scheme to use when writing transformed kernels
+                  to file.
+        :rtype: str
+        '''
+        return self._kernel_naming
+
+    @kernel_naming.setter
+    def kernel_naming(self, value):
+        '''
+        Setter for how to re-name kernels when writing transformed kernels
+        to file.
+
+        :param str value: one of VALID_KERNEL_NAMING_SCHEMES.
+        :raises ValueError: if the supplied value is not a recognised \
```

Comment:
Firstly, I think it is right that this is a value error rather than a configurationerror but I thought I'd ask anyway seeing as most errors in this file are the latter.

Secondly, this error is not trapped in generate.py where the value is set. The posibility of an incorrect input should be handled there.

Reply:
Oops, well spotted. I've added a catch for the `ValueError` (which I chose to raise since it's an error in setting a value in the Configuration object) and then raise a `GenerationError` to signal that the call to `generate()` has failed. I've also added a test for this.